beps/bep_0003.rst

```diff
@@ -87,70 +92,153 @@ announce
 
 info
   This maps to a dictionary, with keys described below.
+  
+``piece layers``
+  A dictionary of strings. For each file in the file tree that is larger than the piece size
+  it contains one string value.
+  The keys are the merkle roots while the values consist of concatenated hashes
+  of one layer within that merkle tree.
+  The layer is chosen so that one hash covers ``piece length`` bytes.
+  For example if the piece size is 16KiB then the leaf hashes are used.
+  If a piece size of 128KiB is used then 3rd layer up from the leaf hashes is used.
+  Layer hashes which exclusively cover data beyond the end of file,
+  i.e. are only needed to balance the tree, are omitted.
+  All hashes are stored in their binary format.
+  
+  A torrent is not valid if this field is absent, the contained hashes do not match
+  the merkle roots or are not from the correct layer.
```

Comment:
Torrents with both v1 and v2 pieces lists could get pretty bloated. Some sites may want to reduce the size of the torrent file without going all the way to magnet links.

BEP 9 is already a de facto requirement due to the prevalence of magnet links so it doesn't seem like that much of a burden to depend on it. Especially since the alternative for reducing metadata size is magnet links, so you're getting a BEP 9 dependency either way.